File: extract_paths.js
const fs = require('fs');
const path = require('path');

const src = fs.readFileSync(
  path.join(__dirname, "antigravity/resources/mapa-interactivo-cr (1).html"),
  'utf8'
);

// Extract the _paths() return object block
const match = src.match(/_paths\(\)\s*\{return\{([\s\S]*?)\}\s*\}/);
if (!match) { console.error('Could not find _paths()'); process.exit(1); }

const block = match[1];

// Parse each key:'value' pair (values may contain commas so we match key-by-key)
const keys = ['CRA','CRG','CRL','CRH','CRSJ','CRC','CRP'];
const paths = {};

for (const key of keys) {
  const re = new RegExp(`${key}:'([^']*)'`);
  const m = block.match(re);
  paths[key] = m ? m[1] : '';
}

fs.writeFileSync(
  path.join(__dirname, 'data/map-paths.json'),
  JSON.stringify(paths, null, 2),
  'utf8'
);

console.log('map-paths.json written:', Object.keys(paths).map(k => `${k}(${paths[k].length} chars)`).join(', '));

File: extract_paths.py
import re, json, os

base = os.path.dirname(os.path.abspath(__file__))
src_path = os.path.join(base, "antigravity/resources/mapa-interactivo-cr (1).html")

with open(src_path, encoding='utf-8') as f:
    src = f.read()

# Extract the _paths() return block
m = re.search(r"_paths\(\)\s*\{return\{([\s\S]*?)\}\s*\}", src)
if not m:
    print("ERROR: _paths() not found"); exit(1)

block = m.group(1)

keys = ['CRA','CRG','CRL','CRH','CRSJ','CRC','CRP']
paths = {}
for key in keys:
    km = re.search(key + r":'([^']*)'", block)
    paths[key] = km.group(1) if km else ''
    print(f"{key}: {len(paths[key])} chars")

out = os.path.join(base, "data/map-paths.json")
with open(out, 'w', encoding='utf-8') as f:
    json.dump(paths, f, ensure_ascii=False, indent=2)
print("Done:", out)
